src/StreamFactoryInterface.php

```diff
@@ -7,17 +7,32 @@
 interface StreamFactoryInterface
 {
     /**
-     * Create a new stream from a resource or a string.
+     * Create a new stream from a resource.
      *
-     * If the argument is a resource, it MUST be readable and SHOULD be seekable. It MAY be writable.
-     * If the argument is a string, a temporary resource will be created that is writable and seekable.
-     * 
-     * If the arugment is a string it will be interpided as the content of the stream. File names or 
-     * file paths are not supported. 
-     *
-     * @param string|resource $resource
+     * @param resource $resource
      *
      * @return StreamInterface
      */
     public function createStream($resource);
+
+    /**
+     * Create a new stream from a file.
+     *
+     * @param string $path
+     * @param string $mode
+     *
+     * @return StreamInterface
+     */
+    public function createStreamFromFile($path, $mode = 'r');
+
+    /**
+     * Create a new stream from a string.
+     *
+     * String content will be used as the body of the stream.
+     *
+     * @param string $content
+     *
+     * @return StreamInterface
+     */
+    public function createStreamFromString($content);
```

Comment:
"Binary string" is a common phrase. What should the method name/type hint be, if not string?